Tallies/TallyClerks/MGxsClerk_class.f90

```diff
@@ -0,0 +1,868 @@
+
+    ! Loop over all energies and spatial bins
+    do i = 1, N*M
+
+      ! Retrieve reaction rates and flux from memory
+      addr = self % getMemAddress() + self % width * (i - 1) - 1
+      call mem % getResult(scatt, scattStd, addr + SCATT_idx)
+      call mem % getResult(flux,  fluxStd,  addr + FLUX_idx)
+      call mem % getResult(scattProb, scattProbStd, addr + SCATT_EV_idx)
+
+      ! Calculate MG constants, being careful to avoid division by zero
+      ! If flux is zero all cross sections must be set to zero
+      if (flux == ZERO) then
+        scattXS    = ZERO
+        scattXSstd = ZERO
+      else
+        scattXS    = scatt/flux
+        scattXSstd = scattXS * sqrt((scattStd/scatt)**2 + (fluxStd/flux)**2)
+      end if
+
+      ! Loop over outgoing energies to calculate the scattering matrices
+      do j = 1, N
+
+        ! Retrieve results from memory
+        call mem % getResult(P0, P0std, addr + SCATT_EV_idx + j)
+        call mem % getResult(P2, P2std, addr + SCATT_EV_idx + 3*N + j)
+        call mem % getResult(P3, P3std, addr + SCATT_EV_idx + 4*N + j)
+        call mem % getResult(P4, P4std, addr + SCATT_EV_idx + 5*N + j)
+        call mem % getResult(P5, P5std, addr + SCATT_EV_idx + 6*N + j)
+        call mem % getResult(P6, P6std, addr + SCATT_EV_idx + 7*N + j)
+        call mem % getResult(P7, P7std, addr + SCATT_EV_idx + 8*N + j)
+
+        ! Calculate scattering matrices and uncertainties
+        idx = N*(i-1) + j
+        if (P0 == ZERO) then
+          P2_res(1:2,idx) = ZERO
+          P3_res(1:2,idx) = ZERO
+          P4_res(1:2,idx) = ZERO
+          P5_res(1:2,idx) = ZERO
+          P6_res(1:2,idx) = ZERO
+          P7_res(1:2,idx) = ZERO
+        else
+          P2_res(1,idx) = P2/scattProb*scattXS
+          P2_res(2,idx) = P2_res(1,idx) * sqrt((P2std/P2)**2 + (scattProbStd/scattProb)**2 + (scattXSstd/scattXS)**2)
+          P3_res(1,idx) = P3/scattProb*scattXS
+          P3_res(2,idx) = P3_res(1,idx) * sqrt((P3std/P3)**2 + (scattProbStd/scattProb)**2 + (scattXSstd/scattXS)**2)
+          P4_res(1,idx) = P4/scattProb*scattXS
+          P4_res(2,idx) = P4_res(1,idx) * sqrt((P4std/P4)**2 + (scattProbStd/scattProb)**2 + (scattXSstd/scattXS)**2)
+          P5_res(1,idx) = P5/scattProb*scattXS
+          P5_res(2,idx) = P5_res(1,idx) * sqrt((P5std/P5)**2 + (scattProbStd/scattProb)**2 + (scattXSstd/scattXS)**2)
+          P6_res(1,idx) = P6/scattProb*scattXS
+          P6_res(2,idx) = P6_res(1,idx) * sqrt((P6std/P6)**2 + (scattProbStd/scattProb)**2 + (scattXSstd/scattXS)**2)
+          P7_res(1,idx) = P7/scattProb*scattXS
+          P7_res(2,idx) = P7_res(1,idx) * sqrt((P7std/P7)**2 + (scattProbStd/scattProb)**2 + (scattXSstd/scattXS)**2)
+        end if
+
+      end do
+
+    end do
+
+  end subroutine processPN
+
+  !!
+  !! Write contents of the clerk to output file
+  !!
+  !! See tallyClerk_inter for details
+  !!
+  !! NOTE: the energy groups are ordered from thermal to fast (G_end -> G_1)
```

Comment:
Any possibility we can flip this round? Just because index 1 being the fastest group is so ubiquitous when doing postprocessing/using other codes. It would just take traversing the do loops in the opposite direction I think

Reply:
Done!